### PR TITLE
Codable routes

### PR DESCRIPTION
**Pull request by nhardman**
feat: add codable routing generation

### CODE REVIEW

**Comment by christiancompton** [comment]
@nhardman @EnriqueL8 Is this ready to go into production?

File: refresh/templates/fromswagger/Routes.swift.hbs [hunk 1]
```diff
@@ -17,14 +17,143 @@ import Kitura
...
+        {{#ifequal this.variant "all"}}
+    func deleteAll_{{this.handlerName}}(completion: (RequestError?) -> Void ) -> Void {
+        completion(.notImplemented)
+    }
+        {{/ifequal}}
+        {{#ifequal this.variant "id"}}
+    func deleteOne_{{this.handlerName}}(id: {{this.idtype}}, completion: (RequestError?) -> Void ) -> Void {
+        completion(.notImplemented)
+    }
+
```
Comment (EnriqueL8) [inline refresh/templates/fromswagger/Routes.swift.hbs:38]: is this space intentional?

File: refresh/templates/fromswagger/Routes.swift.hbs [hunk 2]
```diff
@@ -17,14 +17,143 @@ import Kitura
...
+        {{/ifequal}}
+
+      {{/if}}
+    {{/ifequal}}
+    {{#ifequal this.method "patch"}}
+      {{#if this.codable}}
+    func patchOne_{{this.handlerName}}(id: {{this.idtype}}, patch: {{this.param}}, completion: ({{this.param}}?, RequestError?) -> Void ) -> Void {
+        completion(nil, .notImplemented)
+    }
+
```
Comment (EnriqueL8) [inline refresh/templates/fromswagger/Routes.swift.hbs:69]: same here

File: refresh/templates/fromswagger/Routes.swift.hbs [hunk 3]
```diff
@@ -17,14 +17,143 @@ import Kitura
...
+  {{#if hascodable}}
+    // Codable routes are declared below:
+    //
+  {{/if}}
+  {{#each routedetail}}
+{{#if this.codable}}
+    {{#ifequal this.method "delete"}}
+      {{#ifequal this.variant "all"}}
+    app.router.{{this.method}}("{{#if ../basepath}}\(basePath){{/if}}{{this.route}}", handler: deleteAll_{{this.handlerName}})
```
Comment (EnriqueL8) [inline refresh/templates/fromswagger/Routes.swift.hbs:105]: I believe we should verify the BasePath before

File: refresh/templates/fromswagger/Routes.swift.hbs (hunk 2 above)
Comment (nhardman) [reply refresh/templates/fromswagger/Routes.swift.hbs:69]: fixed

File: refresh/templates/fromswagger/Routes.swift.hbs (hunk 1 above)
Comment (nhardman) [reply refresh/templates/fromswagger/Routes.swift.hbs:38]: fixed

File: refresh/templates/fromswagger/Routes.swift.hbs (hunk 3 above)
Comment (nhardman) [reply refresh/templates/fromswagger/Routes.swift.hbs:105]: Enrique and I agree this should not be changed

**Review by EnriqueL8 — APPROVED** [review]

**Pull request merged**